ProgramLauncher.c

```
@@ -0,0 +1,391 @@
+      }
+   }
+   free(groups);
+   return found;
+}
+#endif
+
+static bool ProgramLauncher_canTrustExecStat(const struct stat* sb) {
+   if (sb->st_uid == geteuid())
+      return S_ISREG(sb->st_mode) && (sb->st_mode & S_IXUSR);
+   if (sb->st_gid == getegid() || group_member(sb->st_gid))
+      return S_ISREG(sb->st_mode) && (sb->st_mode & S_IXGRP);
```

Cppcheck (2.20.0)

[style] The function 'isNonnegative' is never used.

(unusedFunction)